### PR TITLE
Fix import crash

### PR DESCRIPTION
**Pull request by jhsware**
Fixes a couple of issues reported by emil. Reference card "Bugg: TypeError: Cannot read property 'find' of null"

### CODE REVIEW

File: backend/api/importQueue.js [hunk 1]
```diff
@@ -149,14 +149,20 @@ async function resetQueueForImport(courseId) {
...
       },
     });
   } catch (err) {
+    log.warn(
+      "resetQueueForImport failet with error. If this happens once in isolation it is okay, this method is called every time we start an import.",
+      err?.stack
+    );
     throw new Error("Error removing finished entries");
+  } finally {
+    await dbClient.close();
```
Comment (exacs) [inline backend/api/importQueue.js:165]: I think this is the actual bug fix. Keep it! Good catch!

File: backend/api/router.js [hunk 2]
```diff
@@ -153,7 +153,10 @@ router.post("/courses/:id/import/start", async (req, res, next) => {
     );
   }
 
-  await resetQueueForImport(courseId);
+  await resetQueueForImport(courseId).catch(() => {
+    // If the queue hasn't been created the app might crash so lets accept a failure here
+    log.info("Resetting import queue failed but we will try to import anyway.");
+  });
```
Comment (exacs) [inline backend/api/router.js:159]: I don't agree with this. The code had a bug and it's ok to crash the app in such cases. This only silents the bug, does not fix anything

File: backend/api/listAllExams.js [hunk 3]
```diff
@@ -111,11 +111,9 @@ async function listStudentsWithExamsInCanvas(courseId, ladokId) {
...
   const ladokId = await getLadokId(courseId);
   const allScannedExams = await listScannedExams(courseId, ladokId);
-  const studentsWithExamsInCanvas = await listStudentsWithExamsInCanvas(
-    courseId,
-    ladokId
-  );
-  const examsInImportQueue = await getEntriesFromQueue(courseId);
+  const studentsWithExamsInCanvas =
+    (await listStudentsWithExamsInCanvas(courseId, ladokId)) || [];
+  const examsInImportQueue = (await getEntriesFromQueue(courseId)) || [];
```
Comment (exacs) [inline backend/api/listAllExams.js:116]: I don't agree here. If some of the functions `listStudentsWithExamsInCanvas`, etc return a non-list value, then we should fix the root cause and eventually propagate the exception, not hide the problem.

**Review by exacs — CHANGES_REQUESTED** [review]

File: backend/api/router.js (hunk 2 above)
Comment (jhsware) [reply backend/api/router.js]: Fixed in  687f2b01

File: backend/api/listAllExams.js (hunk 3 above)
Comment (jhsware) [reply backend/api/listAllExams.js:116]: I am convinced!

**Review by exacs — APPROVED** [review]

**Pull request merged**